plugin.audio.podcasts/addon.py

```diff
@@ -0,0 +1,369 @@
+
+            if settings.getSetting("opml_file_%i" % g) == "":
+                continue
+
+            path = os.path.join(
+                addon_dir, settings.getSetting("opml_file_%i" % g))
+            opml_data = self._load_opml(path)
+            try:
+                entries = []
+                for o in opml_data["opml"]["body"]["outline"]:
+                    if o["@type"] == "rss":
+                        entries += [{
+                            "path": "opml_file_%i_%i" % (g, len(entries)),
+                            "name": o["@title"],
+                            "params": [
+                                {
+                                    "call": "renderRss",
+                                    "url": o["@xmlUrl"]
+                                }
+                            ],
+                            "node": []
+                        }]
+
+                groups += [{
+                    "path": "opml_file_%i" % g,
+                    "name": opml_data["opml"]["head"]["title"],
+                    "node": entries
+                }]
+
+            except:
+                xbmc.log("Cannot parse opml file %s" % path, xbmc.LOGERROR)
+                pass
+
+        # build groups acc. settings
+        for g in range(self._GROUPS):
+
+            if settings.getSetting("group_%i_enable" % g) == "false":
+                continue
+
+            entries = []
+            for e in range(self._ENTRIES):
+
+                if settings.getSetting("group_%i_rss_%i_enable" % (g, e)) == "false":
+                    continue
+
+                entries += [{
+                    "path": "pod_%i" % e,
+                    "name": settings.getSetting("group_%i_rss_%i_name"
+                                                % (g, e)),
+                    "icon": settings.getSetting("group_%i_rss_%i_icon"
+                                                % (g, e)),
+                    "params": [
+                        {
+                            "call": "renderRss",
+                            "url": settings.getSetting("group_%i_rss_%i_url"
+                                                       % (g, e))
+                        }
+                    ],
+                    "node": []
+                }]
+
+            groups += [{
+                "path": "group_%i" % g,
+                "name": settings.getSetting("group_%i_name" % g),
+                "node": entries
+            }]
+
+        self._menu = [
+            {  # root
+                "path": "",
+                "node": groups
+            }
+        ]
+
+    def _load_opml(self, path):
+
+        try:
+            with open(path) as _opml_file:
+                _data = _opml_file.read()
+                return xmltodict.parse(_data)
+        except:
+            xbmc.log("Cannot open opml file", xbmc.LOGERROR)
+            return None
+
+    def _requestHttp(self, url, redirectCounter=0):
+
+        if redirectCounter >= self._MAX_REDIRECTS:
```

Comment:
I guess this works, but all of this could be taken care of by the `script.module.requests` module.